pkg/metricsadapter/controller.go

```diff
@@ -99,6 +99,7 @@ func nodeTransformFunc(obj interface{}) (interface{}, error) {
 			Name:              node.Name,
 			Namespace:         node.Namespace,
 			Labels:            node.Labels,
+			ResourceVersion:   node.ResourceVersion,
```

Comment:
Do you think this helps? 

I don't have a test, but also ignored the resource version at 
https://github.com/karmada-io/karmada/blob/adef1e59748e1e1d31cb75fffe406b5dd69a66d7/pkg/util/fedinformer/transform.go#L67-L75

Reply:
What about this question?

Reply:
I've already updated the description in the PR.

Since a `resourceVersion` of 0 has special meaning, it seems to trigger a brand new `watch` operation, rather than continuing the previous watch operation.

Reply:
Is there any basis for this? 
If there is an issue here, it is likely a serious one, so it would be better to have concrete evidence.

Reply:
Is there any basis for this? 
If there is an issue here, it is likely a serious one, so it would be better to have concrete evidence.

Reply:
https://github.com/karmada-io/karmada/blob/99d413052ed389133e517637f7f54be401c6ae28/vendor/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L939-L952

Here, you can see, when the resourceVersion is 0, the `cache` will let the watchers re-queue and go through a new round of watching.


I'm not sure if I'm understanding this correctly. 😂

Reply:
Additionally, both `client-go` and `controller-runtime` have extensive special handling for cases where the `resourceVersion` is 0. I believe that forcibly assigning a resourceVersion of 0 is clearly unreasonable.

Reply:
I'll try to look into it. 
But before that, do you think splitting the flags part from this PR is a good idea?

Reply:
ok